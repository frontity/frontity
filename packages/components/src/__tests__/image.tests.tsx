import React from "react";
import TestRenderer from "react-test-renderer";
import Image from "../components/image";

describe("Image", () => {
  test("works just fine", () => {
    const props = {
      alt: "Some fake alt text",
      src: "https://fake-src.com/fake-image.jpg",
      srcSet:
        "https://fake-src.com/fake-image.jpg?w=300 300w, https://fake-src.com/fake-image.jpg?w=150 150w",
      className: "fake-class-name"
    };

    const result = TestRenderer.create(<Image {...props} />).toJSON();
    expect(result).toMatchSnapshot();
  });
});
