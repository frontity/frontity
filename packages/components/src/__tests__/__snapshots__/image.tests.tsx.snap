// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image works just fine 1`] = `
Array [
  <noscript>
    <img
      alt="Some fake alt text"
      className="frontity-lazy-image fake-class-name"
      loading="auto"
      src="https://fake-src.com/fake-image.jpg"
      srcSet="https://fake-src.com/fake-image.jpg?w=300 300w, https://fake-src.com/fake-image.jpg?w=150 150w"
    />
  </noscript>,
  <img
    alt="Some fake alt text"
    className="frontity-lazy-image fake-class-name"
    data-src="https://fake-src.com/fake-image.jpg"
    data-srcset="https://fake-src.com/fake-image.jpg?w=300 300w, https://fake-src.com/fake-image.jpg?w=150 150w"
    loading="auto"
    style={
      Object {
        "visibility": "hidden",
      }
    }
  />,
]
`;
